src/kt_variant.h

```diff
@@ -196,37 +194,34 @@ namespace ktvariant {
     }
 
     static void append_object(SharedBuffer* des, Object* ptr) {
-        //TODO/4.0: rework object cpp -> jvm
 
         // TODO : Investigate on nullable management of Godot. Is Object the only nullable type ?
         if (!ptr) {
             to_kvariant_fromNIL(des, ptr);
             return;
         }
 
-        StringName class_name{ptr->get_class_name()};
-
-        if (!TypeManager::get_instance().JAVA_ENGINE_TYPES_CONSTRUCTORS.has(class_name)) {
-            class_name = ClassDB::get_parent_class(class_name);
-
-            while (class_name == String()) {
-                if (!TypeManager::get_instance().JAVA_ENGINE_TYPES_CONSTRUCTORS.has(class_name)) {
-                    class_name = ClassDB::get_parent_class(class_name);
-                } else {
-                    break;
-                }
-            }
-        }
-
-        bool is_ref{ptr->is_ref_counted()};
-        uint64_t id;
-        if (is_ref) {
-            auto* ref = reinterpret_cast<RefCounted*>(ptr);
-            id = RefDB::get_instance().get_ref_id(ref);
-
-        } else {
-            id = ptr->get_instance_id();
-        }
+		//Create a binding if it doesn't exist yet.
+		KotlinBinding* binding = KotlinBindingManager::get_instance_binding(ptr);
+
+		//We only need to retrieve the constructor if the binding is not ready yet.
+		int classID = 0;
```

Comment:
I think `constructorId` would be more clear

Reply:
Will change it